src/coord/src/coord/dataflow_builder.rs

```diff
@@ -106,6 +106,10 @@ impl<'a> DataflowBuilder<'a> {
             let index_oracle = self.index_oracle();
             let mut valid_indexes = index_oracle.indexes_on(*id).peekable();
             if valid_indexes.peek().is_some() {
+                // Deduplicate indexes by keys, in case we have redundant indexes.
```

Comment:
Ideally we would want to prevent a `CREATE INDEX` statement from even succeeding in the first place if an identical index already exists. If this invariant is upheld, it may make index selection and optimization slightly easier across the entire optimization stack. Unfortunately, we may have data out there that already has duplicate identical indexes (not very plausible, but technically possible).

Reply:
I think .. they might push back on that idk. You could for example `ALTER INDEX` the indexes independently to set different compaction policies. I don't personally mind either way, but .. probably up for them to decide.